libraries/iio/iio.c

```diff
@@ -806,12 +829,17 @@ ssize_t iio_unregister(struct iio_desc *desc,
  * 			UART).
  * @return SUCCESS in case of success or negative value otherwise.
  */
-ssize_t iio_init(struct iio_desc **desc, struct iio_server_ops *iio_server_ops)
+ssize_t iio_init(struct iio_desc **desc, struct iio_init_param *init_param)
 {
-	struct iio_desc *ldesc;
-	struct tinyiiod_ops *ops = (struct tinyiiod_ops *)calloc(1,
-				   sizeof(struct tinyiiod_ops));
+	int32_t			ret;
+	struct iio_desc		*ldesc;
+	struct tinyiiod_ops	*ops;
 
+	if (!init_param || init_param->phy_type == USE_NETWORK)
```

Comment:
why iio_init would fail when you use network physical layer ?

Reply:
Because it is not implemented yet

Reply:
Here you're supposed to check for init_param.

init_param->phy_type is checked at this line and you should add an else if (init_param->phy_type == USE_NETWORK) at this line:
https://github.com/analogdevicesinc/no-OS/blob/87a1ac9e1eea2f501b00ecb9a10c43340154c4f2/libraries/iio/iio.c#L874

Reply:
true, I removed the first check